sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -3,14 +3,113 @@
+     */
+    public AmqpMessageBodyType getBodyType() {
+        return bodyType;
+    }
+
+    /**
+     * Gets an {@link IterableStream} of byte array containing only first byte array set on this
+     * {@link AmqpMessageBody}. This library only support one byte array at present, so the returned list will have only
+     *  one element.
+     * <p><b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method. Get methods not
+     * corresponding to the type of the body throws exception.</b></p>
+     *
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return data set on {@link AmqpMessageBody}.
+     *
+     * @throws IllegalArgumentException If {@link AmqpMessageBodyType} is not {@link AmqpMessageBodyType#DATA DATA}.
+     */
+    public IterableStream<byte[]> getData() {
+        if (bodyType != AmqpMessageBodyType.DATA) {
+
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "This method can only be called for AMQP Data body type at present. Track this issue, "
+                    + "https://github.com/Azure/azure-sdk-for-java/issues/17614 for other body type support in "
+                    + "future."));
+        }
+
+        return new IterableStream<>(data);
+    }
+
     /**
-     * Type representing various supported amqp body types.
+     * Gets first byte array set on this {@link AmqpMessageBody}. This library only support one byte array on Amqp
+     * Message. Look for future releases where we will support multiple byte array and you can use
+     * {@link AmqpMessageBody#getData()} API.
+     * <p><b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method. Get methods not
+     * corresponding to the type of the body throws exception.</b></p>
      *
-     * @return The {@link AmqpBodyType}.
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return data set on {@link AmqpMessageBody}.
+     *
+     * @throws IllegalArgumentException If {@link AmqpMessageBodyType} is not {@link AmqpMessageBodyType#DATA DATA}.
+     * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format" target="_blank">
+     *     Amqp Message Format.</a>
      */
-    AmqpBodyType getBodyType();
+    public byte[] getFirstData() {
+        if (bodyType != AmqpMessageBodyType.DATA) {
+
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "This method can only be called for AMQP Data body type at present. Track this issue, "
+                    + "https://github.com/Azure/azure-sdk-for-java/issues/17614 for other body type support in "
+                    + "future."));
+        }
+
+        return data.get(0);
```

Comment:
If the common case is to have users call this getFirstData API, we shouldn't wrap into a collection and call get(0) on it, as it is just extra work. We should only wrap the first time the getData call is made, and keep it as a byte[] otherwise.